File: lonely-integer/lonely-integer.py
#https://www.hackerrank.com/challenges/three-month-preparation-kit-lonely-integer/problem?isFullScreen=true&h_l=interview&playlist_slugs%5B%5D=preparation-kits&playlist_slugs%5B%5D=three-month-preparation-kit&playlist_slugs%5B%5D=three-month-week-two


#1
def lonelyinteger(a):
    # Write your code here
    seen = {}
    for num in a:
        if num in seen:
            seen[num] = seen[num] + 1
        else:
            seen[num] = 1
    for (key, value) in seen.items():
        if value == 1:
            return key

File: lonely-integer/lonely-integer.js
//1
function lonelyinteger(a) {
    // Write your code here
    const seen = new Map();
    for (let i = 0; i < a.length; i++) {
        if (seen.has(a[i])) {
            seen.set(a[i], seen.get(a[i]) + 1)
        } else {
            seen.set(a[i], 1)
        }
    };
    for (const [key, value] of seen.entries()) {
        if (value === 1) {
            return key
        }
    }
}